### PR TITLE
266: Introduce `Collection.has()` method.

### LINKED ISSUE #266: Introduce collection.has
I know you can use:

```js
collection.getIndex( item ) > 0
```
Or:
```js
!!collection.get( item )
```

But nothing is as pretty as:
```js
if( collection.has( item ) ) {
    //...
}
```

### PR DESCRIPTION
**Pull request by pjasiun**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: introduce `Collection.has()` method. Closes #266.

### CODE REVIEW

File: src/collection.js [hunk 1]
```diff
@@ -209,6 +209,29 @@ export default class Collection {
...
 	}
 
+	/**
+	 * Returns a boolean indicating whether the collection contains an item with the specified id or index.
+	 *
+	 * @param {String|Number} idOrIndex The item id or index in the collection.
+	 * @returns {Boolean} `true` if the collection contains the item, `false` otherwise.
+	 */
+	has( idOrIndex ) {
+		let item;
```
Comment (jodator) [inline src/collection.js]: unused :P

Comment (oskarwrobel) [inline src/collection.js]: Unused.

Comment (pjasiun) [reply src/collection.js]: Outdated. Too late.

File: src/collection.js [hunk 2]
```diff
@@ -209,6 +209,29 @@ export default class Collection {
...
+
+		if ( typeof idOrIndex == 'string' ) {
```
Comment (jodator) [inline src/collection.js]: What if I want to check if `collection.has( item )` - looks handy also.

File: src/collection.js [hunk 3]
```diff
@@ -209,6 +209,29 @@ export default class Collection {
...
+			return this._itemMap.has( idOrIndex );
+		} else if ( typeof idOrIndex == 'number' ) {
```
Comment (jodator) [inline src/collection.js]: I'm not sure if checking index is a way to go. I'd expect something similar to `[].includes()` with additions for our needs.

File: src/collection.js (hunk 1 above)
Comment (jodator) [reply src/collection.js]: Oh, I see :disappointed: you fixed it already.

**Comment by pjasiun** [comment]
> 2\. Apart of that I see that `collection.has( id )` might be useful but I'm not convinced about `collecion.has( index )` since AFAICS the collection keep track of proper index on adding item so one can check if `index > collection.length` to check if that index can be used. Also the semantics of `collection.has( index )` looks odd.

Agree. It's my bad, I was sure that all methods in this class which support `id`, handle also `index`, but then I realized that it is not true. Fixed.

**Pull request merged**